node/pkg/watchers/hedera/watcher.go

```diff
@@ -0,0 +1,612 @@
+					continue
+				}
+				txBody, err := io.ReadAll(resp.Body)
+				if err != nil {
+					logger.Error("query tx response read error", zap.String("network", networkName), zap.Error(err))
+					resp.Body.Close()
+					continue
+				}
+				resp.Body.Close()
+
+				txJSON := string(txBody)
+				logger.Info("RO: txJSON", zap.String("txJSON", txJSON))
+
+				events, err := TxMessageToEvent(logger, txJSON)
+
+				for _, ev := range events {
+					watcher.msgChan <- ev
+					// messagesConfirmed.WithLabelValues(networkName).Inc()
+				}
+			}
+		}
+	}()
+
+	// This function is for normal watcher operations
+	go func() {
+		defer close(errC)
+		t := time.NewTicker(5 * time.Second)
+		client := &http.Client{
+			Timeout: time.Second * 5,
+		}
+		var beginningTS = "1234567890.000000400" // TODO:  Should this be Time.now()?
+		// var logString = "api/v1/contracts/" + contractId + "/results/logs?order=asc&timestamp=gte%3A1234567890.000000400"
+		// var logString = "api/v1/contracts/" + watcher.contract + "/results/logs?order=asc&timestamp=gte%3A" + beginningTS
+
+		for {
+			<-t.C
+			logString := "api/v1/contracts/" + watcher.contract + "/results/logs?order=asc&timestamp=gt%3A" + beginningTS
+			// logString := "api/v1/contracts/" + watcher.contract + "/results/logs?order=asc" + "&topic0=" + TOPIC_LOG_MSG + "&timestamp=[gt%3A&" + beginningTS + "][lt%3A&1671140260.000000400]"
+			logger.Info("current logString", zap.String("string", logString))
+			// Query and report height and set currentSlotHeight
+			resp, err := client.Get(fmt.Sprintf("%s/%s", watcher.urlLCD, logString))
+			if err != nil {
+				logger.Error("query latest logs response error", zap.String("network", networkName), zap.Error(err))
+				continue
+			}
+			logsBody, err := io.ReadAll(resp.Body)
+			if err != nil {
+				logger.Error("query latest logs response read error", zap.String("network", networkName), zap.Error(err))
+				errC <- err
+				resp.Body.Close()
+				continue
+			}
+			resp.Body.Close()
+
+			var events []*common.MessagePublication
+			logJSON := string(logsBody)
+			// logger.Info("logJSON", zap.String("logJSON", logJSON))
+			logs := gjson.Get(logJSON, "logs")
+			// logger.Info("after gjson.get", zap.Stringer("logs", logs))
+			logs.ForEach(func(logKey, logValue gjson.Result) bool {
+				// logger.Info("YIKES..............", zap.Stringer("topics", gjson.Get(logValue.String(), "topics")))
+				topics := gjson.Get(logValue.String(), "topics")
+				topics.ForEach(func(topicKey, topicValue gjson.Result) bool {
+					if topicValue.String() == TOPIC_LOG_MSG {
+						event, err := LogMessageToEvent(logger, logValue.String())
+						// Check for event being nil
+						if err == nil {
+							events = append(events, event)
+							blockNum := gjson.Get(logValue.String(), "block_number")
+							logger.Info("Found True Log Msg", zap.Stringer("block Number", blockNum))
+						}
+						return false // break out of the ForEach() loop
```

Comment:
Do you want to break out if err ==  nil, or should you check for more events?

Reply:
Found the topic of interest.  Break out regardless.